app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+            sample: string
+                sample name to use for all new samples
+                (NOTE: overwrite the name in samples parameter)
+            kwargs_list: bool
+                If set to true, then kwargs with list values are distributed over the samples,
+                otherwise the kwargs are passed as is to the sample.
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            Array with the DB samples
+        '''
+        return super().create_samples(samples, collection=collection, db_create=db_create, **kwargs)
+
+
+class AssetFactory(FactoryBase):
+    # pylint: disable=arguments-differ
+    factory_name = 'asset'
+    sample_class = AssetSample
+
+    def create_sample(self, item, name=None, db_create=False, **kwargs):
+        '''Create an Asset data sample
+
+        Args:
+            item: Item
+                Item model object in which to create an Asset
+            name: string
+                Data name, if not given it creates a 'asset-n' with n being incremented
+                after each function call
+            sample: string
+                sample based on the sample named found in tests.sample_data
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            The data sample
+        '''
+        return super().create_sample(name, item=item, db_create=db_create, **kwargs)
+
+    def create_samples(self, samples, item, db_create=False, **kwargs):
+        '''Creates several Asset samples
+
+        Args:
+            samples: integer | [string]
+                number of DB sample to create or list of sample data name
+            db_create: bool
+                create the sample in the DB
+            sample: string
+                sample name to use for all new samples
+                (NOTE: overwrite the name in samples parameter)
+            kwargs_list: bool
+                If set to true, then kwargs with list values are distributed over the samples,
+                otherwise the kwargs are passed as is to the sample.
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            Array with the DB samples
+        '''
+        return super().create_samples(samples, item=item, db_create=db_create, **kwargs)
+
+
+class Factory:
+    '''Factory for data samples
+    '''
+
+    def __init__(self):
+        self.collections = CollectionFactory()
+        self.items = ItemFactory()
+        self.assets = AssetFactory()
+
+    def create_collection_sample(self, db_create=False, required_only=False, **kwargs):
+        '''Create a collection data sample
+
+        Args:
+            name: string
+                Data name, if not given it creates a 'collection-n' with n being incremented
+                after each function call
+            sample: string
```

Comment:
OK, this one is special: `sample` is obsolete, therefore `required_only` is missing in the docstring here